src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/client/oasysApi/OasysApiService.kt

```diff
@@ -0,0 +1,53 @@
+package uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.client.oasysApi
+
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.client.AuthorisableActionResult
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.client.ClientResult
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.client.oasysApi.model.OasysOffenceDetailWrapper
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.common.exception.NotFoundException
+
+@Service
+class OasysApiService(val oasysApiClient: OasysApiClient) {
+  fun getAssessmentId(prisonerId: String): Long? {
+    val assessments = when (val result = oasysApiClient.getAssessments(prisonerId)) {
+      is ClientResult.Failure -> {
+        throw NotFoundException("No assessment found for prisoner id: $prisonerId")
+      }
+      is ClientResult.Success -> AuthorisableActionResult.Success(result.body)
+    }
+
+    // get the most recent completed assessment
+    val assessment =
+      assessments.entity
+        .timeline
+        .filter { it.status == "COMPLETE" }
+        .sortedByDescending { it.completedDate }
+        .firstOrNull()
+
+    return if (assessment == null) {
+      log.warn("No completed assessment found for prison id $prisonerId")
+      null
+    } else {
```

Comment:
Thoughts on the below change, please?

val assessment = assessments.entity.timeline
    .filter { it.status == "COMPLETED" }
    .maxByOrNull { it.completedDate }

assessment?.let {
    return it.assessmentPk
}

log.warn("No completed assessment found for prison id $prisonerId")
return null

Reply:
MaxByOrNull takes a non nullable date and it is a nullable field so wouldn't work in this case